### PR TITLE
Generate Single command + .devcontainer.json and more

### PR DESCRIPTION
**Pull request by alneberg**
- A command to generate a single report.
 - A Dockerfile and a .devcontainer.json to use docker for development
 - Moved some logic from orderportal to ngi_data.py
 - Recents keep 5 statuses, not 5 dates.
 - Removed the `.data` attribute of `ProjectDataRecord` to  a lot more transparent and less flexible
 - Prettier messed up the template... will try to fix later.

### CODE REVIEW

File: daily_read/ngi_data.py [hunk 1]
```diff
@@ -185,7 +189,16 @@ class ProjectDataRecord(object):
     Raises ValueError if orderer is not present in data, if data is given
     """
 
-    def __init__(self, relative_path, data=None):
+    dates_prio = {
```
Comment (aanil) [inline daily_read/ngi_data.py:192]: We should probably have a single source of the priority, could modify `STATUS_PRIORITY` from main. Something to do in the future.

File: daily_read/order_portal.py [hunk 2]
```diff
@@ -68,71 +68,50 @@ def get_orders(self, node=None, status=None, orderer=None, recent=True):
...
+                        raise ValueError(
+                            f"Multiple reports for Project Progress found in the Order Portal for order {order['identifier']}"
+                        )
+
+            if proj_info.orderer not in order_updates:
+                order_updates[proj_info.orderer] = {
+                    "pull_date": f"{pull_date}",
+                    "active_projects": 0,
+                    "recents": {},
+                    "events": [],
```
Comment (aanil) [inline daily_read/order_portal.py:101]: What will the events be used for?

File: daily_read/order_portal.py [hunk 3]
```diff
@@ -68,71 +68,50 @@ def get_orders(self, node=None, status=None, orderer=None, recent=True):
...
-                for date_value, date_statuses in proj_info.data["proj_dates"].items():
-                    # Activity on 5 recent dates
-                    recents_keys = sorted(order_updates[proj_info.orderer]["recents"].keys())
-                    if len(recents_keys) < 5:
-                        for status in date_statuses:
-                            order_updates[proj_info.orderer]["recents"].setdefault(date_value, []).append(
-                                (proj_info.data["project_name"], status)
-                            )
+            elif order["identifier"] not in self.projects_data.data.keys():
+                log.debug(f"Order portal id: {order['identifier']} not found in data fetched form sources")
```
Comment (aanil) [inline daily_read/order_portal.py]:
```suggestion
                log.debug(f"Order portal id: {order['identifier']} not found in data fetched from sources")
```

File: daily_read/order_portal.py (hunk 2 above)
Comment (alneberg) [reply daily_read/order_portal.py:101]: I extract the recents from the events. So it's just intermediate storage

File: daily_read/ngi_data.py [hunk 4]
```diff
@@ -200,19 +213,68 @@ def __init__(self, relative_path, data=None):
...
-            self.orderer = data["orderer"]
+        # Figure out project status from the latest status(es)
 
-            self.data = data
+        if project_dates:
+            latest_date, latest_statuses = sorted(project_dates.items(), reverse=True)[0]
+
+            # If multiple statuses for the same date, choose the one with highest prio
+            if len(latest_statuses) > 1:
+                status = sorted(latest_statuses, key=lambda s: self.dates_prio[s])[0]
```
Comment (aanil) [inline daily_read/ngi_data.py]: Should it be self.status?

Comment (alneberg) [reply daily_read/ngi_data.py]: Done!

File: daily_read/ngi_data.py (hunk 1 above)
Comment (alneberg) [reply daily_read/ngi_data.py:192]: Created https://github.com/NationalGenomicsInfrastructure/DailyRead/issues/21 to not completely forget this.

**Comment by alneberg** [comment]
I also removed the prettier action now, it was too brittle with the jinja plugin to be blocking the PRs. Can still be useful to have locally though.

**Review by aanil — APPROVED** [review]
👍  Looks good!

**Pull request merged**